### PR TITLE
Use recreate strategy for ironic deployment

### PR DESCRIPTION
**Pull request by lentzi90**
Ironic cannot run with more than 1 replica (the second crashes immediately) but the default rolling update strategy will try to get a second replica up and running before it removes the old one. This means that the rollout always gets stuck unless we first manually scale down the deployment to 0, then update, and finally scale it up again.

The recreate strategy is a better fit since it does the scale down
automatically.

For reference, here is how we currently upgrade ironic in the upgrade tests:
https://github.com/Nordix/metal3-dev-env/blob/1ab2dda79c717083f8ced21cdaeb885c93559dc6/vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml#L217-L257

### CODE REVIEW

**Comment by lentzi90** [comment]
Waiting for https://github.com/metal3-io/baremetal-operator/pull/918 to simplify the upgrade steps for ironic. Once that PR is merged, we can skip the scale down and scale up since it will be done automatically.

/hold

**Comment by lentzi90** [comment]
/hold cancel
(Accidentally added it on the wrong PR...)

**Review by namnx228 — APPROVED** [review]
LGTM, this is a good improvement

**Comment by smoshiur1237** [comment]
Looks good to me.

**Review by macaptain — APPROVED** [review]

File: ironic-deployment/ironic/ironic.yaml
Comment (dtantsur) [inline ironic-deployment/ironic/ironic.yaml:8]: Let's please reword this, ironic can run with a lot of replicates, *we* cannot run it this way.

Comment (lentzi90) [reply ironic-deployment/ironic/ironic.yaml:8]: Right, will do!

Comment (lentzi90) [reply ironic-deployment/ironic/ironic.yaml]: Better?

**Comment by smoshiur1237** [comment]
PR is ready to get approved and lgtm.

**Pull request merged**
